api/wallet_test.go

```diff
@@ -240,6 +325,71 @@ func TestIntegrationWalletGETSiacoins(t *testing.T) {
 	}
 }
 
+// TestIntegrationWalletSweepSeedPOST probes the POST call to
+// /wallet/sweep/seed.
+func TestIntegrationWalletSweepSeedPOST(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	st, err := createServerTester("TestIntegrationWalletSweepSeedPOST")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer st.server.Close()
+
+	// send coins to a new wallet, then sweep them back
+	var key crypto.TwofishKey
+	rand.Read(key[:])
+	w, err := wallet.New(st.cs, st.tpool, filepath.Join(st.dir, "wallet2"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = w.Encrypt(key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = w.Unlock(key)
+	if err != nil {
+		t.Fatal(err)
+	}
+	addr, _ := w.NextAddress()
+	st.wallet.SendSiacoins(types.SiacoinPrecision.Mul64(100), addr.UnlockHash())
+	st.miner.AddBlock()
+
+	seed, _, _ := w.PrimarySeed()
+	seedStr, _ := modules.SeedToString(seed, "english")
+
+	// Sweep the coins we sent
+	var wsp WalletSweepPOST
+	qs := url.Values{}
+	qs.Set("seed", seedStr)
+	err = st.postAPI("/wallet/sweep/seed", qs, &wsp)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Should have swept more than 80 SC
+	if wsp.Coins.Cmp(types.SiacoinPrecision.Mul64(80)) <= 0 {
+		t.Fatalf("swept fewer coins (%v SC) than expected %v+", wsp.Coins.Div(types.SiacoinPrecision), 80)
+	}
+
+	// Add a block so that the sweep transaction is processed
+	st.miner.AddBlock()
+
+	// Sweep again; should find no coins. An error will be returned because
+	// the found coins cannot cover the transaction fee.
+	err = st.postAPI("/wallet/sweep/seed", qs, &wsp)
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	// Call /wallet/sweep/seed with an invalid seed
+	qs.Set("seed", "foo")
+	err = st.postAPI("/wallet/sweep/seed", qs, &wsp)
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
```

Comment:
In the future this should probably be a separate test, as it's polluting this test by having + executing on multiple ideas. Much easier to debug problems with tests if each test has a single thing that it's focused on contributing.